x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_search_preview/api_call_flyout.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+
+import {
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiTab,
+  EuiTabs,
+  EuiTitle,
+} from '@elastic/eui';
+import { SearchRequest, SearchResponse } from '@elastic/search-ui-engines-connector';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { generateEncodedPath } from '../../../../shared/encode_path_params';
+import { EuiLinkTo } from '../../../../shared/react_router_helpers';
+
+import { EngineViewTabs, ENGINE_TAB_PATH } from '../../../routes';
+
+export interface APICallData {
+  request: SearchRequest;
+  response: SearchResponse;
+}
+
+export interface APICallFlyoutProps {
+  engineName: string;
+  lastAPICall: APICallData;
+  onClose: () => void;
+}
+
+export const APICallFlyout: React.FC<APICallFlyoutProps> = ({
+  engineName,
+  onClose,
+  lastAPICall,
+}) => {
+  const [tab, setTab] = useState<'request' | 'response'>('request');
+
+  const contents = JSON.stringify(lastAPICall[tab], null, 2);
+
+  return (
+    <EuiFlyout onClose={onClose} size="m">
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle>
+          <h2>
+            <FormattedMessage
+              id="xpack.enterpriseSearch.content.engine.searchPreivew.apiCallFlyout.title"
+              defaultMessage="API Call"
+            />
+          </h2>
+        </EuiTitle>
```

Comment:
Can we add `<EuiSpacer size="s" />` between the title and tabs to give it a little more space?

Reply:
totes 👜